xbmc/addons/binary-addons/BinaryAddonManager.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ *      Copyright (C) 2005-2017 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Kodi; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "BinaryAddonManager.h"
+
+#include "addons/AddonManager.h"
+#include "utils/log.h"
+
+using namespace ADDON;
+
+CBinaryAddonManager::CBinaryAddonManager()
+{
+
+}
+
+CBinaryAddonManager::~CBinaryAddonManager()
+{
+
+}
+
+bool CBinaryAddonManager::Init()
+{
+  std::vector<std::pair<std::string, bool>> binaryAddonList;
+  if (!CAddonMgr::GetInstance().GetInstalledBinaryAddonPaths(binaryAddonList))
+  {
+    CLog::Log(LOGERROR, "Binary Addons: failed to get list");
+    return false;
+  }
+
+  for (auto addon : binaryAddonList)
+  {
+    BinaryAddonBase base = std::make_shared<CBinaryAddonBase>(addon.first);
+    if (!base->IsUsable())
+      continue;
+    m_installedAddons[base->ID()] = base;
+    if (addon.second)
+      m_enabledAddons[base->ID()] = base;
```

Comment:
what's the advantage of having two vectors here, one for all usable installed addons, another for the usable installed enabled addons? The addons contained in the second vector are 100% subset of the addons in the first vector. would it be more efficient to have just one vector containing the addon along with its 'enabled' flag? => header change: <code>std::vector<std::pair<std::shared_ptr<CBinaryAddonBase>, bool>> m_usableAddons;</code>?